plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/Reporter.kt

```diff
@@ -7,22 +7,23 @@ import org.gradle.api.tasks.JavaExec
 import org.gradle.util.GFileUtils
 import java.io.File
 import java.io.FileOutputStream
+import net.swiftzer.semver.SemVer
 
 /**
  * Supported reporter types.
  */
-enum class ReporterType(val reporterName: String, val availableSinceVersion: String, val fileExtension: String) {
-    PLAIN("plain", "0.9.0", "txt"),
-    PLAIN_GROUP_BY_FILE("plain?group_by_file", "0.9.0", "txt"),
-    CHECKSTYLE("checkstyle", "0.9.0", "xml"),
-    JSON("json", "0.9.0", "json");
+enum class ReporterType(val reporterName: String, val availableSinceVersion: SemVer, val fileExtension: String) {
+    PLAIN("plain", SemVer(0, 9, 0), "txt"),
+    PLAIN_GROUP_BY_FILE("plain?group_by_file", SemVer(0, 9, 0), "txt"),
+    CHECKSTYLE("checkstyle", SemVer(0, 9, 0), "xml"),
+    JSON("json", SemVer(0, 9, 0), "json");
 }
 
 /**
  * Apply reporter to the task.
  */
 fun JavaExec.applyReporter(target: Project, extension: KtlintExtension, sourceSetName: String) {
-    if (isReportAvailable(extension.version, extension.reporter.availableSinceVersion)) {
+    if (SemVer.parse(extension.version).compareTo(extension.reporter.availableSinceVersion) >= 0) {
```

Comment:
Would have done this with `SemVer.parse(extension.version) >= extension.reporter.availableSinceVersion` instead, but sure.